topology.go

```diff
@@ -290,3 +342,69 @@ func GetComponent(ctx context.Context, db *gorm.DB, id string) (*models.Componen
 
 	return &component, nil
 }
+
+type Tag struct {
+	Key string `json:"key"`
+	Val string `json:"val"`
+}
+
+type TopologyResponse struct {
+	componentIDs    []string
+	healthStatusMap map[string]struct{}
+	typeMap         map[string]struct{}
+	tagMap          map[string]struct{}
+
+	Components     models.Components `json:"components,omitempty"`
+	HealthStatuses []string          `json:"healthStatuses,omitempty"`
+	Teams          []string          `json:"teams,omitempty"`
+	Tags           []Tag             `json:"tags,omitempty"`
+	Types          []string          `json:"types,omitempty"`
+}
+
+func (t *TopologyResponse) AddHealthStatuses(s string) {
+	if t.healthStatusMap == nil {
+		t.healthStatusMap = make(map[string]struct{})
+	}
```

Comment:
You can create a `NewTopologyResponse` func which returns a TopologyResponse with components and maps already initialized

Reply:
I chose not to do that since it's very easy to directly instantiate the variable without using a constructor.

Reply:
I think if you use a CTE, you wouldn't have to do it in code, it will be handled by DB instead using `Distinct`

Something like,
```
WITH topology AS (
  <original_query>
)

SELECT json_build_object('components', to_json(topology), 'tags', SELECT DISTINCT(tags) FROM topology, ... and so on) FROM topology;
```